x/tokenfactory/types/params.go

```diff
@@ -105,3 +113,24 @@ func validateFeeCollectorAddress(i interface{}) error {
 
 	return nil
 }
+
+func validateWhitelistedHooks(i interface{}) error {
+	hooks, ok := i.([]*HookWhitelist)
+	if !ok {
+		return fmt.Errorf("invalid parameter type: %T", i)
+	}
+
+	seenHooks := map[string]bool{}
+	for _, hook := range hooks {
+		hookStr := hook.String()
+		if seenHooks[hookStr] {
+			return fmt.Errorf("duplicate whitelisted hook: %#v", hook)
```

Comment:
what does `%#v` do? why not `%s` with `hookStr` as value?

Reply:
It is supposed to pretty-print it. But actually comparing the output I agree, hookStr is much clearer